### PR TITLE
Tweaks to get tests running, added CICD file

### PR DESCRIPTION
**Pull request by JR-USAF**

### CODE REVIEW

File: .github/workflows/pytest_and_ruff.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest
+        pip install .
```
Comment (JP-USAF) [inline .github/workflows/pytest_and_ruff.yml:43]: Does this need [extra]? I am not familiar with this option and some quick googling didn't yield any answers. I just noticed that it is being used everywhere else.

Comment (JR-USAF) [reply .github/workflows/pytest_and_ruff.yml:43]: Good catch, this was an artifact left over from copying the original code over. I removed it.

**Comment by AM-USAF** [comment]
Do we also want to add a version check?

File: .github/workflows/pytest_and_ruff.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest
+        pip install .[extras]
```
Comment (AM-USAF) [inline .github/workflows/pytest_and_ruff.yml:27]: remove extras

**Review by AM-USAF — APPROVED** [review]

**Pull request merged**